include/zarray/zdispatcher.hpp

```diff
@@ -12,31 +12,78 @@
 
 #include <xtl/xmultimethods.hpp>
 
+#include "xtensor/xstrided_view.hpp"
 #include "zdispatching_types.hpp"
 #include "zmath.hpp"
 
 namespace xt
 {
     namespace mpl = xtl::mpl;
 
-    template <class type_list>
+    template <class type_list, class undispatched_type_list = mpl::vector<>>
     using zrun_dispatcher_impl = xtl::functor_dispatcher
     <
         type_list,
         void,
+        undispatched_type_list,
         xtl::static_caster,
         xtl::basic_fast_dispatcher
     >;
 
-    template <class type_list>
+    template <class type_list, class undispatched_type_list = mpl::vector<>>
     using ztype_dispatcher_impl = xtl::functor_dispatcher
     <
         type_list,
         size_t,
+        undispatched_type_list,
         xtl::static_caster,
         xtl::basic_fast_dispatcher
     >;
 
+    /********************
+     * zview_dispatcher *
+     ********************/
+
+    // The view dispatcher dispatches on the single argument and forwards a
+    // single undispatched argument (the slices).
+
+    template <class F>
+    class zview_dispatcher
```

Comment:
Should it be more generic, with a template for the undispatched argument type? Currently it is hard-coded in `zrun_dispatcher`.

Reply:
Yes I think it could make sense, this way we could reuse the code for the reducers for instance.

Reply:
But then the type of the undispatched argument should be a template of the class? Because `m_run_dispatcher` depends on this type.

Reply:
yes